code/modules/power/rbmk/rbmk.dm

```diff
@@ -0,0 +1,958 @@
+	var/no_coolant_ticks = 0	//How many times in succession did we not have enough coolant? Decays twice as fast as it accumulates.
+	var/datum/looping_sound/rbmk/alarm
+
+//Use this in your maps if you want everything to be preset.
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/preset
+	id = "default_reactor_for_lazy_mappers"
+
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/destroyed
+	icon_state = "reactor_slagged"
+	slagged = TRUE
+	vessel_integrity = 0
+
+
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/Destroy()
+	QDEL_NULL(alarm)
+	. = ..()
+
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/examine(mob/user)
+	. = ..()
+	if(Adjacent(src, user))
+		if(do_after(user, 1 SECONDS, target=src))
+			var/percent = vessel_integrity / initial(vessel_integrity) * 100
+			var/msg = "<span class='warning'>The reactor looks operational.</span>"
+			switch(percent)
+				if(0 to 10)
+					msg = "<span class='boldwarning'>[src]'s seals are dangerously warped and you can see cracks all over the reactor vessel! </span>"
+				if(10 to 40)
+					msg = "<span class='boldwarning'>[src]'s seals are heavily warped and cracked! </span>"
+				if(40 to 60)
+					msg = "<span class='warning'>[src]'s seals are holding, but barely. You can see some micro-fractures forming in the reactor vessel.</span>"
+				if(60 to 80)
+					msg = "<span class='warning'>[src]'s seals are in-tact, but slightly worn. There are no visible cracks in the reactor vessel.</span>"
+				if(80 to 90)
+					msg = "<span class='notice'>[src]'s seals are in good shape, and there are no visible cracks in the reactor vessel.</span>"
+				if(95 to 100)
+					msg = "<span class='notice'>[src]'s seals look factory new, and the reactor's in excellent shape.</span>"
+			. += msg
+
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/fuel_rod))
+		if(power >= 20)
+			to_chat(user, "<span class='notice'>You cannot insert fuel into [src] when it has been raised above 20% power.</span>")
+			return FALSE
+		if(fuel_rods.len >= 5)
+			to_chat(user, "<span class='warning'>[src] is already at maximum fuel load.</span>")
+			return FALSE
+		to_chat(user, "<span class='notice'>You start to insert [W] into [src]...</span>")
+		radiation_pulse(src, temperature)
+		if(do_after(user, 5 SECONDS, target=src))
+			if(!length(fuel_rods))
+				start_up() //That was the first fuel rod. Let's heat it up.
+			fuel_rods += W
+			W.forceMove(src)
+			radiation_pulse(src, temperature) //Wear protective equipment when even breathing near a reactor!
+		return TRUE
+	if(!slagged && istype(W, /obj/item/sealant))
+		if(power >= 20)
+			to_chat(user, "<span class='notice'>You cannot repair [src] while it is running at above 20% power.</span>")
+			return FALSE
+		if(vessel_integrity >= 350)
```

Comment:
```suggestion
		if(vessel_integrity >= 0.875*initial(vessel_integrity))
```
Seems a bit weird to base some stuff on the initial vessel integrity and other stuff on absolute values, better to standardize.